Modules/+Sources/Msquared.m

```diff
@@ -0,0 +1,481 @@
+                                                                    'help_text', 'Apply fine tuning to the resonator in units of percent of total range (0 -> 200 V).');
+        resonator_voltage = Prefs.Double(NaN,   'units', 'V',   'readonly', true, ...
+                                                                    'help_text', 'The amount of fine tuning upon the resonator. Interact with this via resonator_percent.');
+
+        % PULSEBLASTER prefs
+        PB_host =           Prefs.String(Sources.Msquared.no_server, 'set', 'set_PB_host');
+        PB_line =           Prefs.Integer(1, 'min', 1, 'set', 'set_PB_line', 'help_text', 'Indexed from 1.');
+    end
+    
+    methods(Access=private)
+        function obj = Msquared()
+            obj.loadPrefs;      % This will call set_hwserver_host, which will instantiate the hardware.
+            obj.getFrequency(); % Update prefs based on current values.
+        end
+    end
+    methods(Static)
+        function obj = instance()                   % Standard instance method to prevent duplication.
+            mlock;
+            persistent Object
+            if isempty(Object) || ~isvalid(Object)
+                Object = Sources.Msquared();
+            end
+            obj = Object;
+        end
+    end
+    
+    methods     % Helper methods for laser/wavemeter communication/datapolling.
+        function reply = com(obj, fn, varargin)     % Helper function for hwsever communication (fills in moduleName and deals with reserved clients).
+            if isempty(obj.moduleName); return; end
+            
+            try
+                reply = obj.hwserver.com(obj.moduleName, fn, varargin{:});
+            catch err
+                if contains(err.message, 'Another client was using')
+                    % Grab line with exception in it
+                    exception = strsplit(err.message, newline);
+                    mask = cellfun(@(a)startswith(a, 'Exception: '), exception);
+                    exception = exception{mask};
+                    answer = questdlg(sprintf('%s\nDo you want to override the other client?', exception),...
+                        mfilename, 'Yes', 'No', 'No');
+                    if strcmp(answer,'Yes')
+                        % Override and recall
+                        obj.com('force_client');
+                        reply = obj.hwserver.com(obj.moduleName, fn, varargin{:});
+                        return
+                    else
+                        error(exception);
+                    end
+                end
+                rethrow(err);
+            end
+        end
+        
+        function callGetWavelength(obj)             % Calls the solstis method 'get_wavelength' and fills prefs in.
+            if isempty(obj.hwserver) || ~isvalid(obj.hwserver) || isempty(obj.moduleName)
+                obj.status = obj.no_server;
+                obj.tuning = false;
+%                 obj.wavelength = NaN;
```

Comment:
Remove commented-out lines

Reply:
That's a reminder that those properties are returned, so I would like to keep them around.